src/imgix-core-js.js

```diff
@@ -36,8 +42,24 @@
         this.settings[key] = val;
       }
 
-      if (!this.settings.host) {
-        throw new Error('ImgixClient must be passed a valid host');
+      if(!Array.isArray(this.settings.domains)){
+        this.settings.domains = [this.settings.domains]
+      }
+
+      if (!this.settings.host && this.settings.domains.length === 0) {
+        throw new Error('ImgixClient must be passed valid domain(s)');
+      }
+
+      if (this.settings.shard_strategy !== SHARD_STRATEGY_CRC
+          && this.settings.shard_strategy !== SHARD_STRATEGY_CYCLE) {
+        throw new Error('Shard strategy must be one of ' +
+          SHARD_STRATEGY_CRC + ', ' + SHARD_STRATEGY_CYCLE);
+      }
+
+      if(this.settings.host){
+        // console.warn("'host' argument is deprecated; use 'domains' instead.")
```

Comment:
I left it commented by mistake.

I didn't use `console.error` because `host` is still a valid (but deprecated) argument. Browser consoles highlight `console.warn` and `console.warn` messages but not `console.log`.

Reply:
Yea, browsers handle it nicely but in Node `console.warn` [is an alias for `console.error`](https://nodejs.org/api/console.html#console_console_warn_data_args). Given that this prints to `stderr`, I think it's probably safer to avoid it and use `console.log`.

Reply:
Isn't the warning messages being redirected to `stderr` a good thing? You can easily redirect `stderr` to a file or `/dev/null` without polluting the library-importing-code's `console.log` output messages.

Reply:
Yea, you're right. Best to leave it as-is.